WordPress/Classes/ViewRelated/Media/MediaPickerMenu.swift

```diff
@@ -0,0 +1,272 @@
+                    }
+                }
+            }
+        } else {
+            DDLogError("No image representation available for provider with registered types: \(provider.registeredTypeIdentifiers)")
+            DispatchQueue.main.async {
+                completion(nil, nil)
+            }
+        }
+    }
+}
+
+// MARK: - MediaPickerMenu (Camera)
+
+protocol ImagePickerControllerDelegate: AnyObject {
+    // Hides `NSObject` and `UINavigationControllerDelegate` conformances that
+    // the original `UIImagePickerControllerDelegate` has.
+
+    /// - parameter info: If the info is empty, nothing was selected.
+    func imagePicker(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any])
+}
+
+extension MediaPickerMenu {
+    /// Returns an action from capturing media using the device's camera.
+    ///
+    /// - parameters:
+    ///   - camera: The camera to use. By default, `.rear`.
+    ///   - delegate: The delegate.
+    func makeCameraAction(
+        camera: UIImagePickerController.CameraDevice = .rear,
+        delegate: ImagePickerControllerDelegate
+    ) -> UIAction {
+        UIAction(
+            title: cameraActionTitle,
+            image: UIImage(systemName: "camera"),
+            attributes: [],
+            handler: { _ in showCamera(camera: camera, delegate: delegate) }
+        )
+    }
+
+    private var cameraActionTitle: String {
+        guard let filter else {
+            return Strings.takePhotoOrVideo
+        }
+        switch filter {
+        case .images: return Strings.takePhoto
+        case .videos: return Strings.takeVideo
+        }
+    }
+
+    func showCamera(camera: UIImagePickerController.CameraDevice, delegate: ImagePickerControllerDelegate) {
+        switch AVCaptureDevice.authorizationStatus(for: .video) {
+        case .authorized, .notDetermined:
+            actuallyShowCamera(camera: camera, delegate: delegate)
+        case .restricted, .denied:
+            showAccessRestrictedAlert()
+        @unknown default:
+            showAccessRestrictedAlert()
+        }
+    }
+
+    private func actuallyShowCamera(camera: UIImagePickerController.CameraDevice, delegate: ImagePickerControllerDelegate) {
+        let picker = UIImagePickerController()
+        picker.sourceType = .camera
+        picker.cameraDevice = camera
+        picker.videoQuality = .typeHigh
+        if let filter {
+            switch filter {
+            case .images: picker.mediaTypes = [UTType.image.identifier]
+            case  .videos: picker.mediaTypes = [UTType.movie.identifier]
+            }
+        } else {
+            picker.mediaTypes = [UTType.image.identifier, UTType.movie.identifier]
+        }
+
+        let delegate = ImagePickerDelegate(delegate: delegate)
+        picker.delegate = delegate
+        objc_setAssociatedObject(picker, &MediaPickerMenu.strongDelegateKey, delegate, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
Q: are you doing this to ensure the picker / delegate gets deallocated along with the menu?

Reply:
I need to retain `ImagePickerDelegate` that gets created by this method. It gets deallocated along with the image picker.